hibernate-core/src/main/java/org/hibernate/sql/ast/tree/delete/DeleteStatement.java

```diff
@@ -27,10 +27,12 @@ public class DeleteStatement extends AbstractMutationStatement {
 
 	public static final String DEFAULT_ALIAS = "to_delete_";
 	private final Predicate restriction;
+	private final Predicate subqueryRestriction;
```

Comment:
This should go away and the predicate for the cleanup subquery should be built on demand.